### PR TITLE
Login redirect

### PR DESCRIPTION
**Pull request by Minglee01**
@incuna/js please review.

Redirect user to logged out page when their session expires.

### CODE REVIEW

File: src/providers/auth-login-redirect-token-auth-clear.js [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+                }
+            });
+
+            this.routeIsAnonymous = function (route) {
+                // By default, all routes should be anonymous.
+                var routeIsAnonymous = true;
+                if (route && route.anonymous === false) {
+                    routeIsAnonymous = false;
+                }
+                return routeIsAnonymous;
```
Comment (wytrych) [inline src/providers/auth-login-redirect-token-auth-clear.js]: I would simplify this like so:
```
if (!route) {
    return false;
}
return route.anonymous;
```

File: tests/unit/loginRedirectTokenAuthClear-spec.js [hunk 2]
```diff
@@ -0,0 +1,140 @@
...
+
+                    this.TOKEN_AUTH_MOCK.LOGOUT_REDIRECT_URL = '/?show-login';
+
+                    inject(function ($location) {
+                        this.$location = $location;
+                    });
+                    this.$location.url('/current/');
+
+                    // Run.
+                    this.onTokenAuthClear.redirect();
```
Comment (wytrych) [inline tests/unit/loginRedirectTokenAuthClear-spec.js]: I think this line should be inside the test

**Comment by wytrych** [comment]
Looks good

**Comment by Minglee01** [comment]
@wytrych updated

File: tests/unit/loginRedirectTokenAuthClear-spec.js [hunk 3]
```diff
@@ -53,28 +53,6 @@
...
-
-                it('should return false for a non-anonymous route', function () {
-                    expect(this.onTokenAuthClear.routeIsAnonymous(this.nonAnonymousRoute)).toBe(false);
-                });
-
-                it('should return true for an anonymous route', function () {
-                    expect(this.onTokenAuthClear.routeIsAnonymous(this.anonymousRoute)).toBe(true);
-                });
-
-            });
```
Comment (wytrych) [inline tests/unit/loginRedirectTokenAuthClear-spec.js]: I would keep these test.

File: tests/unit/loginRedirectTokenAuthClear-spec.js [hunk 4]
```diff
@@ -85,8 +63,6 @@
                         this.$location = $location;
                     });
                     this.$location.url('/current/');
-
-                    // Run.
                     this.onTokenAuthClear.redirect();
```
Comment (wytrych) [inline tests/unit/loginRedirectTokenAuthClear-spec.js]: This line should go to the test, not be in `beforeEach` I think

File: tests/unit/loginRedirectTokenAuthClear-spec.js (hunk 3 above)
Comment (Minglee01) [inline tests/unit/loginRedirectTokenAuthClear-spec.js]: because of the alteration routeIsAnonymous variable no longer exists

**Comment by Minglee01** [comment]
@wytrych not sure how to fix the test.

File: dist/angular-token-auth.js [hunk 5]
```diff
@@ -438,10 +438,11 @@
 
             this.routeIsAnonymous = function (route) {
                 // By default, all routes should be anonymous.
-                if (!route) {
-                    return false;
```
Comment (wytrych) [inline dist/angular-token-auth.js]: The logic was inverted sorry. Should be 
```
if (!route) {
    return true;
}
return route.anonymous;
```

Comment (henrahmagix) [inline dist/angular-token-auth.js]: How about
```js
return !route || route.anonymous;
```

File: src/providers/auth-login-redirect-token-auth-clear.js (hunk 1 above)
Comment (henrahmagix) [inline src/providers/auth-login-redirect-token-auth-clear.js]: I think being explicit here is better: this function returns a boolean. Also the `route &&` part isn't checking if `route` is defined, it's just catching the situation where it's not so it doesn't TypeError on `route.anonymous`.

File: tests/unit/angularTokenAuth-spec.js [hunk 6]
```diff
@@ -9,18 +9,6 @@
...
 
-        beforeEach(function () {
-
-            // Inject dependencies. module() can no longer be called after this.
-            inject(function ($rootScope) {
-                // Inject something to create the injector.
-                // Uncomment this line if you are having test troubles. The
-                // injector won't run if there are missing dependencies.
-                // console.log('generic: If this log does not show, there are missing dependencies.');
-            });
```
Comment (henrahmagix) [inline tests/unit/angularTokenAuth-spec.js]: This is required for `it('should load the dependencies without error')` to run below. No specs will run without an injector being created.

File: src/providers/auth-login-redirect-token-auth-clear.js (hunk 1 above)
Comment (Minglee01) [inline src/providers/auth-login-redirect-token-auth-clear.js]: im keeping with being explicit as its just a pure copy of whats in another one of our repos.

**Comment by Minglee01** [comment]
@henrahmagix  please check again thanks

**Comment by henrahmagix** [comment]
👍

**Pull request merged**